clippy_lints/src/methods/mod.rs

```diff
@@ -2428,8 +2453,10 @@ fn check_methods<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>, msrv: Optio
             },
             ("expect", [_]) => match method_call(recv) {
                 Some(("ok", [recv], _)) => ok_expect::check(cx, expr, recv),
+                Some(("err", [recv], span)) => err_expect::check(cx, expr, recv, msrv, span),
```

Comment:
Here you only pass the span from `err`, ideally you want the span from `err().expect` to get that you can use:

```suggestion
                Some(("err", [recv], err_span)) => err_expect::check(cx, expr, recv, msrv, span, err_span),
```

The `span` value belongs to the `expect()` and is defined in line 2388

https://github.com/rust-lang/rust-clippy/blob/6206086dd5a83477131094b1f0ef61b10e7ced42/clippy_lints/src/methods/mod.rs#L2388

Then you can use `err_span.to(expect_span)` in `err_expect.rs`. See [`Span::to`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/struct.Span.html#method.to) :upside_down_face:

Reply:
> Here you only pass the span from `err`, ideally you want the span from `err().expect` to get that you can use:
> 
> The `span` value belongs to the `expect()` and is defined in line 2388

Sorry I didn't understand that part

Reply:
Currently, this code only span the suggestion over the `err` from `err().expect(..)`. If you look at the `.fixed` file you can see that this results in the following replacement:

```diff
- test_debug.err().expect("Testing debug type");    
+ test_debug.expect_err().expect("Testing debug type");
```

While we actually want to replace the `err()` and `expect` part. The Span object defines the region of code that should be replaced in the source file. The variable `span` is already defined in line 2388 and belongs to the `expect` from `err().expect(...)`. You can join the span from `err` and `expect` together with `err_span.to(expect_span)`. Is that a bit better explained? :upside_down_face: